lib/s.ml

```diff
@@ -64,6 +64,34 @@ end
 module type SANDBOX = sig
   type t
 
+  val pp : t Fmt.t
+  (** A pretty-printer for sandbox environments *)
+
+  type config [@@deriving sexp]
+  (** The type of sandbox configurations *)
+
+  val cmdliner : config Cmdliner.Term.t 
+  (** [cmdliner] is used for command-line interfaces to generate the necessary flags 
+      and parameters to setup a specific sandbox's configuration. *)
+
+  val create : ?state_dir:string -> config -> t Lwt.t   
```

Comment:
Also, not sure that `config` and `create` should be in `SANDBOX`. I see this already caused some trouble for the mock sandbox. This could stay in `runc_sandbox.mli` for now, and then when there are multiple implementations, we can have this in a shared `sandbox.mli`.